src/libs/shim-proto/README.md

```diff
@@ -0,0 +1,40 @@
+# shim-proto
+
+Protobuf message and ttRPC service definitions for Containerd shim v2 protocol.
+
+## Design
+
+The `shim-proto` crate provides [Protobuf](https://github.com/protocolbuffers/protobuf.git) message
+and [ttRPC](https://github.com/containerd/ttrpc.git) service definitions for the
+[Containerd shim v2](https://github.com/containerd/containerd/blob/main/runtime/v2/task/shim.proto) protocol.
+Details about `shim-v2` can be found [here](https://github.com/kata-containers/kata-containers/tree/main/docs/design/architecture#shim-v2-architecture)
+
+The message and service definition source files are auto-generated from the
+[shim-v2 protobuf source file](https://github.com/containerd/containerd/blob/main/runtime/v2/task/shim.proto)
+by using [ttrpc-codegen](https://github.com/containerd/ttrpc-rust/tree/master/ttrpc-codegen). So please do not
+edit those auto-generated source files. If upgrading/modifications are needed, please follow the steps:
```

Comment:
idea: having this as an script src/libs/shim-proto/scripts/update_generated_code.sh